sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamJoinRel.java

```diff
@@ -295,4 +303,37 @@ private BeamRecord buildNullRow(BeamRelNode relNode) {
 
     return new Pair<>(leftIndex, rightIndex);
   }
+
+  private PCollection<BeamRecord> joinAsLookup(BeamRelNode leftRelNode, BeamRelNode rightRelNode,
+      PCollectionTuple inputPCollections, BeamSqlEnv sqlEnv) throws Exception {
+    PCollection<BeamRecord> factStream = leftRelNode.buildBeamPipeline(inputPCollections, sqlEnv);
+    BeamSqlSeekableTable seekableTable = getSeekableTableFromRelNode(rightRelNode, sqlEnv);
+
+    return factStream.apply("join_as_lookup",
+        new BeamJoinTransforms.JoinAsLookup(condition, seekableTable,
+            CalciteUtils.toBeamRowType(rightRelNode.getRowType()),
+            CalciteUtils.toBeamRowType(leftRelNode.getRowType()).getFieldCount()));
+  }
+
+  private BeamSqlSeekableTable getSeekableTableFromRelNode(BeamRelNode relNode, BeamSqlEnv sqlEnv) {
+    BeamIOSourceRel srcRel = (BeamIOSourceRel) relNode;
+    String tableName = Joiner.on('.').join(srcRel.getTable().getQualifiedName());
```

Comment:
`Joiner.on('.').join` here seems not needed? since you have passed one string into it.

Reply:
Actually `srcRel.getTable().getQualifiedName()` returns a `List`, in Calcite there's a concept of `database` which we don't touch so far.